cli/src/pcluster/templates/budget_builder.py

```diff
@@ -0,0 +1,71 @@
+from aws_cdk import aws_budgets as budgets
+from aws_cdk.core import Construct
+
+from pcluster.config.cluster_config import BaseClusterConfig
+
+
+class CostBudgets:
+    """Create the budgets based on the configuration file."""
+
+    def __init__(self, scope: Construct, cluster_config: BaseClusterConfig):
+        self.cluster_config = cluster_config
+        self.scope = scope
+
+    def create_budgets(self):
+        """Create the Cfn templates for the list of budgets."""
+        budget_list = self.cluster_config.dev_settings.budgets
+        cfn_budget_list = []
+        for index, budget in enumerate(budget_list):
+
+            budget_data = budgets.CfnBudget.BudgetDataProperty(
+                budget_name=f"CfnBudget{self.cluster_config.cluster_name}" + str(index),
```

Comment:
If the budget name only contains cluster_name and index, we can not recognize whether the budget is queue budget, custom budget or cluster budget from its name. Let's make the budget_name more clear

Reply:
Good idea. I made it more descriptive